frontends/web/src/routes/new-settings/components/dropdowns/activecurrenciesdropdown.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import Select, { ActionMeta, DropdownIndicatorProps, OptionProps, components } from 'react-select';
+import { selectFiat, unselectFiat, SharedProps } from '../../../../components/rates/rates';
+import { Fiat } from '../../../../api/account';
+import { SelectedCheckLight } from '../../../../components/icon';
+import dropdownStyles from './dropdowns.module.css';
+import activeCurrenciesDropdownStyle from './activecurrenciesdropdown.module.css';
+
+type SelectOption = {
+    label: String;
+    value: Fiat;
+  }
+
+type TSelectProps = {
+    options: SelectOption[];
+  } & SharedProps;
+
+
+// a multi-select dropdown
+export const ActiveCurrenciesDropdown = ({ options, active, selected }: TSelectProps) => {
+  const [selectedCurrencies, setSelectedCurrencies] = useState<SelectOption[]>([]);
+  const [search, setSearch] = useState('');
+
+  useEffect(() => {
+    if (selected.length > 0) {
+      const formattedSelectedCurrencies = selected.map(currency => ({ label: currency, value: currency }));
+      setSelectedCurrencies(formattedSelectedCurrencies);
+    }
+  }, [selected]);
+
+  const DropdownIndicator = (props: DropdownIndicatorProps<SelectOption, true>) => {
+    return (
+      <components.DropdownIndicator {...props}>
+        <div className={dropdownStyles.dropdown} />
+      </components.DropdownIndicator>
+    );
+  };
+
+  const Option = (props: OptionProps<SelectOption, true>) => {
+    const { label, value } = props.data;
+    const selected = selectedCurrencies.findIndex(currency => currency.value === value) >= 0;
+    return (
+      <components.Option {...props}>
+        <span>{label}</span>
+        {selected ? <SelectedCheckLight /> : null}
+      </components.Option>
+    );
+  };
+  return (
+    <Select
+      className={`
+         ${dropdownStyles.select}
+         ${activeCurrenciesDropdownStyle.select}
+         ${search.length > 0 ? activeCurrenciesDropdownStyle.hideMultiSelect : ''}
+         `}
+      classNamePrefix="react-select"
+      isSearchable
+      isClearable={false}
+      components={{ DropdownIndicator, IndicatorSeparator: () => null, MultiValueRemove: () => null, Option }}
+      isMulti
+      closeMenuOnSelect={false}
+      hideSelectedOptions={false}
+      value={[...selectedCurrencies].reverse()}
```

Comment:
reverse() here so the latest selected currency will be displayed at the most front of the string.

E.g: 
1. Let's say selected currencies looks like this: **GBP, EUR**
2. User selects **AUD**
3. Now, selected looks like this: **AUD, GBP, EUR**

Reply:
ahh that makes sense, I was wondering what the order is and why it changes before. This way it always shows the newest nice 👍